### PR TITLE
Update filetype_extensions.conf (Assembly)

### PR DESCRIPTION
**Pull request by DevynCJohnson**
Added various Assembly src file-extensions.

Assembly code has a variety of extensions that indicate a particular target system. For instance, ".a51" and ".asm51" contain Assembly code specifically for the 8051 microprocessor.

### CODE REVIEW

File: data/filetype_extensions.conf [hunk 1]
```diff
@@ -7,7 +7,7 @@ Abc=*.abc;*.abp;
 ActionScript=*.as;
 Ada=*.adb;*.ads;
 Asciidoc=*.asciidoc;*.adoc;
-ASM=*.asm;
+ASM=*.asm;*.asm51;*.a51;*.asm;*.ll;*.s03;*.s79;*.s82;*.s90;*.s;
```
Comment (b4n) [inline data/filetype_extensions.conf]: `*.asm` is duplicated.
I can't find anything using `*.ll` as an assembly extension.  Seems to be use by [Lex](https://en.wikipedia.org/wiki/Lex_%28software%29) and some obscure web stuff or something, but nothing about assembly.

`*.S` might be a nice addition, too.

**Comment by b4n** [comment]
Does all those extensions refer to assembly dialects the ASM filetype actually supports?  There is like a gazillion dialects, and they often are completely different, so it needs to actually be compatible.

File: data/filetype_extensions.conf (hunk 1 above)
Comment (DevynCJohnson) [inline data/filetype_extensions.conf]: Whoops. I bet that duplicate mistake does not give me a good first-impression on this team. (^u^)

*.ll is LLVM assembly.

From now on, I will be discuss my intended commits and ideas on the Geany-dev mailing list before submitting contributions. Hopefully, that will save Geany's GitHub team some time and inconvenience. I will also include more detailed commit notes and be more detailed in explanations.

**Comment by DevynCJohnson** [comment]
@b4n , true, there are numerous assembly extensions. Geany does support 8051 assembly, so I added those extensions (asm51 & a51) since I use them sometimes.

**Comment by elextr** [comment]
The lexer supports a sort of combined NASM/MASM assembler format (according to the comment) and doesn't care what the target chip is so it should be fairly flexible.

**Pull request closed without merging**